src/nvim/os/shell.c

```diff
@@ -207,16 +207,15 @@ static int do_os_system(char **argv,
   char prog[MAXPATHL];
   xstrlcpy(prog, argv[0], MAXPATHL);
 
-  Stream in, out, err;
   LibuvProcess uvproc = libuv_process_init(&main_loop, &buf);
   Process *proc = &uvproc.process;
   MultiQueue *events = multiqueue_new_child(main_loop.events);
   proc->events = events;
   proc->argv = argv;
-  proc->in = input != NULL ? &in : NULL;
-  proc->out = &out;
-  proc->err = &err;
-  int status = process_spawn(proc);
+  int status = process_spawn(proc, input != NULL, true, true);
+  proc->in.events = events;
+  proc->out.events = events;
+  proc->err.events = events;
```

Comment:
Changing `proc->{in,out,err}.events` from `NULL` to `events` should not be necessary.